collector/textfile.go

```diff
@@ -218,6 +219,27 @@ func (c *textFileCollector) Update(ch chan<- prometheus.Metric) error {
 			mtime, families, err := c.processFile(path, f.Name(), ch)
 
 			for _, mf := range families {
+				// Check for metrics with inconsistent help texts. In such a case:
+				// * update `node_textfile_scrape_error`,
+				// * update the last help text set with the differentiating help text (for each such instance), and,
+				// * log an error.
+				// Additionally, generate metric only for the first such instance (i.e., *mf.Help == helpTexts[0]).
+				if helpTexts, seen := metricsNamesToHelpTexts[*mf.Name]; seen {
+					if helpTexts[0] != *mf.Help || helpTexts[1] != "" {
+						metricsNamesToHelpTexts[*mf.Name] = [2]string{helpTexts[0], *mf.Help}
+						errored = true
+						level.Error(c.logger).Log("msg", "inconsistent metric help text",
+							"metric", *mf.Name,
+							"original_help_text", helpTexts[0],
+							"new_help_text", *mf.Help,
+							// This conditional prevents more than one inconsistent help text metric file paths from
+							// being associated to that metric, i.e., only the first file path will be recorded in case
+							// of two or more metrics with inconsistent help texts.
```

Comment:
This comment seems out of place, in the middle of the error logging.

Reply:
Reworded, I added this in to justify why only the first path is being logged.